view/model/index.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  Scene,
+  PerspectiveCamera,
+  WebGLRenderer,
+  MeshBasicMaterial,
+  BufferAttribute,
+  BufferGeometry,
+  Vector3,
+  Points,
+  Camera
+} from 'three'
+import { Rgb, P3, Rec2020 } from 'culori'
+import { TrackballControls } from 'three/examples/jsm/controls/TrackballControls'
+
+import { oklch, lch } from '../../lib/colors.js'
+import { showP3, showRec2020 } from '../../stores/settings.js'
+
+interface ModelData {
+  coordinates: Vector3[]
+  colors: Rgb['r'][] | Rgb['g'][] | Rgb['b'][]
+}
+
+let scene: Scene
+let camera: Camera
+let renderer: WebGLRenderer
+let controls: TrackballControls
+
+let canvas = document.querySelector<HTMLCanvasElement>('.model_canvas')!
+
+function init(): void {
+  scene = new Scene()
+  camera = new PerspectiveCamera(75, 3, 0.1, 1000)
+  renderer = new WebGLRenderer({
+    canvas
+  })
+
+  renderer.setPixelRatio(window.devicePixelRatio)
+  renderer.setSize(350, 210)
+  renderer.setClearColor(0xffffff, 0)
+  LCH ? camera.position.setZ(50) : camera.position.setZ(1)
+  LCH ? camera.position.setX(300) : camera.position.setX(2)
+  LCH ? camera.position.setY(200) : camera.position.setY(1)
+  controls = new TrackballControls(camera, renderer.domElement)
+  LCH ? (controls.minDistance = 400) : (controls.minDistance = 2)
+  controls.maxDistance = controls.minDistance
+}
+
+function getModelData(): ModelData {
+  let coordinates = []
+  let colors = []
+
+  let mode: Rgb['mode'] | P3['mode'] | Rec2020['mode'] = 'rgb'
+  if (showRec2020.get()) {
+    mode = 'rec2020'
+  } else if (showP3.get()) {
+    mode = 'p3'
+  }
+
+  for (let x = 0.01; x <= 1; x += 0.01) {
+    for (let y = 0.01; y <= 1; y += 0.01) {
+      for (let z = 0.01; z <= 1; z += 0.01) {
+        let rgb: Rgb | P3 | Rec2020 = { mode, r: x, g: y, b: z }
+        let color
+        LCH ? (color = lch(rgb)) : (color = oklch(rgb))
```

Comment:
You should use `build()` instead

https://github.com/evilmartians/oklch-picker/blob/main/lib/colors.ts#L44